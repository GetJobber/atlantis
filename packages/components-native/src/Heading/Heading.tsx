import React from "react";
import {
  FontFamily,
  TextAlign,
  TextColor,
  TruncateLength,
  Typography,
  TypographyProps,
} from "../Typography";
import { tokens } from "../utils/design";

type HeadingStyle = Pick<
  TypographyProps<FontFamily>,
  "fontFamily" | "fontWeight" | "size" | "lineHeight" | "color"
>;
type HeadingColor = Extract<TextColor, "text" | "subdued" | "heading">;
export type HeadingLevel = "title" | "subtitle" | "heading" | "subHeading";

interface HeadingProps<T extends HeadingLevel>
  extends Pick<TypographyProps<"base">, "selectable"> {
  /**
   * Text to display.
   */
  readonly children: string;

  /**
   * The type of heading, e.g., "Title"
   */
  readonly level?: T;

  /**
   * The text color of heading
   */
  readonly variation?: HeadingColor;

  /**
   * Uses the reverse variant of the text color for the heading
   */
  readonly reverseTheme?: boolean;

  /**
   * Alignment of heading
   */
  readonly align?: TextAlign;

  /**
   * The maximum amount of lines the text can occupy before being truncated with "...".
   * Uses predefined string values that correspond to a doubling scale for the amount of lines.
   */
  readonly maxLines?: TruncateLength;

  /**
   * Allow text to be resized based on user's device display scale
   */
  readonly allowFontScaling?: boolean;
}

export const HEADING_MAX_SCALED_FONT_SIZE: Record<HeadingLevel, number> = {
  subHeading: tokens["typography--fontSize-base"] * 1.375,
  heading: tokens["typography--fontSize-base"] * 1.5,
  subtitle: tokens["typography--fontSize-base"] * 1.5,
  title: tokens["typography--fontSize-base"] * 2.125,
};

export function Heading<T extends HeadingLevel = "heading">({
  children,
  level,
  variation = "heading",
  reverseTheme = false,
  align,
  maxLines = "unlimited",
  allowFontScaling = true,
  selectable,
}: HeadingProps<T>): JSX.Element {
  const headingStyle = getHeadingStyle(level, variation);
  const accessibilityRole = "header";

  return (
    <Typography
      {...{
        ...headingStyle,
        accessibilityRole,
        reverseTheme,
        align,
        maxLines,
        allowFontScaling,
      }}
      maxFontScaleSize={HEADING_MAX_SCALED_FONT_SIZE[level as HeadingLevel]}
      selectable={selectable}
    >
      {children}
    </Typography>
  );
}

function getHeadingStyle(
  level: HeadingLevel = "heading",
  variation: HeadingColor,
): HeadingStyle {
  const headingLevelToStyle: Record<HeadingLevel, HeadingStyle> = {
    title: {
      fontFamily: "display",
      fontWeight: "extraBold",
      size: "jumbo",
      lineHeight: "jumbo",
      color: variation,
    },
    subtitle: {
      fontFamily: "base",
      fontWeight: "bold",
      size: "largest",
      lineHeight: "largest",
      color: variation,
    },
    heading: {
      fontFamily: "base",
      fontWeight: "bold",
      size: "larger",
      lineHeight: "large",
      color: variation,
    },
    subHeading: {
      fontFamily: "base",
      fontWeight: "semiBold",
      size: "default",
      lineHeight: "base",
      color: variation,
    },
  };

  return headingLevelToStyle[level];
}
