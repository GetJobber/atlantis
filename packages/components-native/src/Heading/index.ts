export { Heading, HEADING_MAX_SCALED_FONT_SIZE } from "./Heading";
export type { HeadingLevel } from "./Heading";
