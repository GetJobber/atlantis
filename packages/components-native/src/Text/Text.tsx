import React from "react";
import {
  BaseWeight,
  LineHeight,
  TextAccessibilityRole,
  TextAlign,
  TextSize,
  TextVariation,
  TruncateLength,
  Typography,
  TypographyProps,
} from "../Typography";
import { tokens } from "../utils/design";
import { TypographyUnsafeStyle } from "../Typography/Typography";

export interface TextProps
  extends Pick<TypographyProps<"base">, "maxFontScaleSize" | "selectable"> {
  /**
   * Visual hierarchy of the text
   */
  readonly level?: TextLevel;

  /**
   * Color variation of text
   */
  readonly variation?: TextVariation;

  /**
   * Change the appearance of the text
   */
  readonly emphasis?: "strong";

  /**
   * Allow text to be resized based on user's device display scale
   */
  readonly allowFontScaling?: boolean;

  /**
   * Determines whether text should be scaled down to fit based on maxLines prop
   */
  readonly adjustsFontSizeToFit?: boolean;

  /**
   * The maximum amount of lines the text can occupy before being truncated with "...".
   * Uses predefined string values that correspond to a doubling scale for the amount of lines.
   */
  readonly maxLines?: TruncateLength;

  /**
   * Alignment of text
   */
  readonly align?: TextAlign;

  /**
   * Text to display
   */
  readonly children?: string;

  /**
   * Reverse theme for better display on dark background
   */
  readonly reverseTheme?: boolean;

  /**
   * Have text styled with strike through
   */
  readonly strikeThrough?: boolean;

  /**
   * Use italic font style
   */
  readonly italic?: boolean;

  /**
   * Underline style to use for the text. The non-solid style is only supported
   * on iOS, as per React Native's Text component's limitations.
   * https://reactnative.dev/docs/text-style-props#textdecorationstyle-ios
   */
  readonly underline?: "solid" | "dotted";

  /**
   * This will make the text inaccessible to the screen reader.
   * This should be avoided unless there is a good reason.
   * For example this is used in InputText to make it so the label isn't
   * selectable because it is already read from the accessibilityLabel
   * of the TextInput
   */
  readonly hideFromScreenReader?: boolean;

  /**
   * **Use at your own risk:** Custom style for specific elements. This should only be used as a
   * **last resort**. Using this may result in unexpected side effects.
   * More information [here](https://atlantis.getjobber.com/storybook/?path=/docs/guides-customizing-components--docs#unsafe_-props).
   */
  readonly UNSAFE_style?: TypographyUnsafeStyle;
}

export type TextLevel = "text" | "textSupporting";

interface LevelStyle {
  readonly size: TextSize;
  readonly lineHeight: LineHeight;
}

const levelStyles: Record<TextLevel, LevelStyle> = {
  text: {
    size: "default",
    lineHeight: "base",
  },
  textSupporting: {
    size: "small",
    lineHeight: "tight",
  },
};

const MAX_TEXT_FONT_SIZE_SCALE = 50;
export const TEXT_MAX_SCALED_FONT_SIZES: Record<TextLevel, number> = {
  text: MAX_TEXT_FONT_SIZE_SCALE,
  textSupporting: tokens["typography--fontSize-base"],
};

export function Text({
  level = "text",
  variation = "base",
  emphasis,
  allowFontScaling = true,
  adjustsFontSizeToFit = false,
  maxLines = "unlimited",
  align,
  children,
  reverseTheme = false,
  strikeThrough = false,
  italic = false,
  hideFromScreenReader = false,
  maxFontScaleSize,
  UNSAFE_style,
  underline,
  selectable,
}: TextProps): JSX.Element {
  const accessibilityRole: TextAccessibilityRole = "text";

  return (
    <Typography
      color={variation}
      UNSAFE_style={UNSAFE_style}
      fontFamily="base"
      fontStyle={italic ? "italic" : "regular"}
      fontWeight={getFontWeight({ level, emphasis })}
      maxFontScaleSize={maxFontScaleSize || TEXT_MAX_SCALED_FONT_SIZES[level]}
      selectable={selectable}
      underline={underline}
      {...{
        ...levelStyles[level],
        allowFontScaling,
        align,
        adjustsFontSizeToFit,
        accessibilityRole,
        reverseTheme,
        maxLines,
        strikeThrough,
        italic,
        hideFromScreenReader,
      }}
    >
      {children}
    </Typography>
  );
}

function getFontWeight({
  level,
  emphasis,
}: Pick<TextProps, "level" | "emphasis">): BaseWeight {
  if (emphasis === "strong") return "semiBold";
  if (level === "textSupporting") return "medium";

  return "regular";
}
