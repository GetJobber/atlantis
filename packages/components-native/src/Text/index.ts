export { Text, TextLevel } from "./Text";
