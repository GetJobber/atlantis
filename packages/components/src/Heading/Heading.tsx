import React, { ReactNode } from "react";
import { Typography, TypographyOptions } from "../Typography";
import { useAtlantisConfig } from "../utils/useAtlantisConfig";

type HeadingLevel = 1 | 2 | 3 | 4 | 5 | 6;

interface HeadingProps {
  /**
   * @default 5
   */
  readonly level: HeadingLevel;
  readonly children: ReactNode;
}

export type LevelMap = Record<HeadingLevel, TypographyOptions>;

export function Heading({ level = 5, children }: HeadingProps) {
  const { JOBBER_RETHEME: inRetheme } = useAtlantisConfig();

  const levelMap: LevelMap = {
    1: {
      element: "h1",
      size: "jumbo",
      fontWeight: inRetheme ? "extraBold" : "black",
      textColor: "heading",
    },
    2: {
      element: "h2",
      size: "largest",
      fontWeight: inRetheme ? "bold" : "black",
      textColor: "heading",
    },
    3: {
      element: "h3",
      size: "larger",
      fontWeight: inRetheme ? "bold" : "extraBold",
      textColor: "heading",
    },
    4: {
      element: "h4",
      size: "large",
      fontWeight: "bold",
      textColor: "heading",
    },
    5: {
      element: "h5",
      size: "base",
      fontWeight: "bold",
      textColor: "heading",
    },
    6: {
      element: "h6",
      size: "small",
      textCase: "uppercase",
      fontWeight: inRetheme ? "semiBold" : "bold",
      textColor: "heading",
    },
  };

  return <Typography {...levelMap[level]}>{children}</Typography>;
}
