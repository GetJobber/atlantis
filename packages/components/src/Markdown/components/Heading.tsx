import React, { PropsWithChildren } from "react";
import { BaseElementProps } from "../BaseElementProps";
import { Heading as InternalHeading } from "../../Heading";

interface HeadingProps extends BaseElementProps, Record<string, any> {
  level: number;
}

export function Heading({
  level,
  children,
  node,
  ...props
}: PropsWithChildren<HeadingProps>) {
  return (
    <InternalHeading level={level as 1 | 2 | 3 | 4 | 5 | 6} {...props}>
      {children}
    </InternalHeading>
  );
}
