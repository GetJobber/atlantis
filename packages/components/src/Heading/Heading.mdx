---
menu: Components
---

import { Playground, Props } from "docz";
import { ComponentStatus } from "@jobber/docx";
import { Heading } from ".";

# Heading

<ComponentStatus stage="ready" responsive="yes" accessible="yes" />

Headings are used as the titles of each major section of a page in the
interface. Reserved for short, important text or numbers.

```ts
import { Heading } from "@jobber/components/Heading";
```

<Playground>
  <Heading level={2}>Subtitle</Heading>
</Playground>

## Design & Usage Guidelines

### Heading 1

This should explain the main subject of the page. There should only be one on a
page.

<Playground>
  <Heading level={1}>New client</Heading>
</Playground>

### Heading 2

Used to categorize large groups of content. For example, there are two main
categories when creating a new client: client details and property details.

This heading can be skipped if there are no large groups that need breaking up.

<Playground>
  <Heading level={2}>Client details</Heading>
  ...
  <Heading level={2}>Property details</Heading>
  ...
</Playground>

### Heading 3

Used to group content and forms on a single topic.

<Playground>
  <Heading level={3}>Contact details</Heading>
  ...
  <Heading level={3}>Notifications</Heading>
  ...
  <Heading level={3}>Additional details</Heading>
  ...
</Playground>

### Heading 4

Used to group contents after a [heading 3](#heading-3) or inside a card
component.

<Playground>
  <Heading level={4}>Phone numbers</Heading>
  ...
  <Heading level={4}>Email</Heading>
  ...
  <Heading level={4}>Pigeon</Heading>
  ...
</Playground>

### Heading 5

Used to group contents after a [heading 4](#heading-4).

<Playground>
  <Heading level={5}>Receives SMS</Heading>
</Playground>

---

## Props

<Props of={Heading} />

---

## Related components

* For introductory text after a page title or paragraph text for body copy, use [Text](text).
