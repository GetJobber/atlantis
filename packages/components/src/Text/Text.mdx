---
menu: Components
---

import { Playground, Props } from "docz";
import { ComponentStatus } from "@jobber/docx";
import { Text } from ".";
import { Content } from "../Content/";

# Text

<ComponentStatus stage="ready" responsive="yes" accessible="yes" />

Text is used to add an additional visual meaning on your content

```ts
import { Text } from "@jobber/components/Text";
```

<Playground>
  <Text>
    "640K is more memory than anyone will ever need on a computer" ‚Äì Not Bill
    Gates
  </Text>
</Playground>

## Props

<Props of={Text} />

---

## Variations

### Default

Use this as the body text

<Playground>
  <Text>Ask the information you need upfront from clients and new leads</Text>
</Playground>

### Subdued

De-emphasize a piece of text that is less important to the user

<Playground>
  <Text variation="subdued">Job note linked to related invoice</Text>
</Playground>

### Feedback message

<Playground>
  <Text variation="success">Invoice sent</Text>
  <br />
  <Text variation="error">Name is required</Text>
  <br />
  <Text variation="warn">Your message is over 160 characters</Text>
  <br />
  <Text variation="info">
    Drag to rearrange the order that fields show up in Jobber
  </Text>
</Playground>

## maxLines

Used to truncate large text with `...` based on line wrapping, such that the
total number of lines does not exceed the preset. This is an optional prop. If
left undefined all the contents are shown.

<Playground>
  <Content>
    <Text maxLines="single">
      This will show 1 line. Lorem ipsum dolor sit amet, consectetur adipiscing
      elit. Vestibulum nec pulvinar nunc. Suspendisse nec eros pretium, rutrum
      purus sit amet, finibus ante. Donec pretium condimentum scelerisque.
    </Text>

    <Text maxLines="small">
      This will show 2 lines. Lorem ipsum dolor sit amet, consectetur adipiscing
      elit. Vestibulum nec pulvinar nunc. Suspendisse nec eros pretium, rutrum
      purus sit amet, finibus ante. Donec pretium condimentum scelerisque. Duis eu
      ligula nec metus suscipit feugiat.Etiam sapien sapien, mattis eu tincidunt
      quis, pretium sed metus. Maecenas quis dolor lacinia libero rhoncus
      fringilla. Cras mi ante, euismod nec tortor in, tempus mollis nulla.
    </Text>

    <Text maxLines="base">
      This will show 4 lines. Lorem ipsum dolor sit amet, consectetur adipiscing
      elit. Vestibulum nec pulvinar nunc. Suspendisse nec eros pretium, rutrum
      purus sit amet, finibus ante. Donec pretium condimentum scelerisque. Duis eu
      ligula nec metus suscipit feugiat.Etiam sapien sapien, mattis eu tincidunt
      quis, pretium sed metus. Maecenas quis dolor lacinia libero rhoncus
      fringilla. Cras mi ante, euismod nec tortor in, tempus mollis nulla.
    </Text>

    <Text maxLines="large">
      This will show 8 lines. Lorem ipsum dolor sit amet, consectetur adipiscing
      elit. Vestibulum nec pulvinar nunc. Suspendisse nec eros pretium, rutrum
      purus sit amet, finibus ante. Donec pretium condimentum scelerisque. Duis eu
      ligula nec metus suscipit feugiat.Etiam sapien sapien, mattis eu tincidunt
      quis, pretium sed metus. Maecenas quis dolor lacinia libero rhoncus
      fringilla. Cras mi ante, euismod nec tortor in, tempus mollis nulla. Etiam
      eu lacus nibh. Donec tristique lacus magna, vulputate tristique lacus
      blandit vitae. Aenean vitae commodo metus. Fusce eu risus quis orci pharetra
      consequat nec nec libero.
    </Text>

    <Text maxLines="larger">
      This will show 16 lines. Lorem ipsum dolor sit amet, consectetur adipiscing
      elit. Vestibulum nec pulvinar nunc. Suspendisse nec eros pretium, rutrum
      purus sit amet, finibus ante. Donec pretium condimentum scelerisque. Duis eu
      ligula nec metus suscipit feugiat.Etiam sapien sapien, mattis eu tincidunt
      quis, pretium sed metus. Maecenas quis dolor lacinia libero rhoncus
      fringilla. Cras mi ante, euismod nec tortor in, tempus mollis nulla. Etiam
      eu lacus nibh. Donec tristique lacus magna, vulputate tristique lacus
      blandit vitae. Aenean vitae commodo metus. Fusce eu risus quis orci pharetra
      consequat nec nec libero. Lorem ipsum dolor sit amet, consectetur adipiscing
      elit. Vestibulum nec pulvinar nunc. Suspendisse nec eros pretium, rutrum
      purus sit amet, finibus ante. Donec pretium condimentum scelerisque. Duis eu
      ligula nec metus suscipit feugiat.Etiam sapien sapien, mattis eu tincidunt
      quis, pretium sed metus. Maecenas quis dolor lacinia libero rhoncus
      fringilla. Cras mi ante, euismod nec tortor in, tempus mollis nulla. Etiam
      eu lacus nibh. Donec tristique lacus magna, vulputate tristique lacus
      blandit vitae. Aenean vitae commodo metus. Fusce eu risus quis orci pharetra
      consequat nec nec libero.
    </Text>

  </Content>
</Playground>

## Sizes

`Text` can be used with different sizes.

### Base

`Base` is the default size and is optimized for the widest range of uses,
including:

- Long paragraphs of text content
- Descriptions of subsequent UI elements
- Labels on UI elements such as checkboxes and radio buttons
- Freeform user-generated content

<Playground>
  <Text size="base">
    Stay organized when creating estimates, work orders, and billing documents.
  </Text>
</Playground>

### Large

Typically `large` is to be used as an introductory paragraph that gives
important information about a page, and requires greater prominence.

<Playground>
  <Text size="large">
    Attract new customers by asking happy clients for a Facebook recommendation.
  </Text>
</Playground>

### Small

The `small` size is intended to be used as helper content on things like
checkboxes, radio buttons, inputs or any other UI elements that need help
provided.

<Playground>
  <Text size="small">What is this, a font for üêús?</Text>
</Playground>
