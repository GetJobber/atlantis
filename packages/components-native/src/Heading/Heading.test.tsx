import React from "react";
import { render } from "@testing-library/react-native";
import { Heading } from "./Heading";
import { Text } from "../Text";

describe("when Heading called with text as the only prop", () => {
  it("should match snapshot", () => {
    const view = render(<Heading>Default Heading</Heading>).toJSON();

    expect(view).toMatchSnapshot();
  });
});

describe("when Heading called with reverseTheme", () => {
  it("should match snapshot", () => {
    const view = render(
      <Heading reverseTheme>Reverse Theme Heading</Heading>,
    ).toJSON();

    expect(view).toMatchSnapshot();
  });
});

describe("when Heading called with title variation", () => {
  it("should match snapshot", () => {
    const view = render(
      <Heading level={"title"}>Title Heading</Heading>,
    ).toJSON();

    expect(view).toMatchSnapshot();
  });
});

describe("when Heading called with Subtitle variation", () => {
  it("should match snapshot", () => {
    const view = render(
      <Heading level={"subtitle"}>Subtitle</Heading>,
    ).toJSON();

    expect(view).toMatchSnapshot();
  });
});

describe("when Heading called with sub-heading variation", () => {
  it("should match snapshot", () => {
    const view = render(
      <Heading level={"subHeading"}>Sub-Heading</Heading>,
    ).toJSON();

    expect(view).toMatchSnapshot();
  });
});

describe("when Heading called with sub-heading variation and text-color", () => {
  it("should match snapshot", () => {
    const view = render(
      <Heading level={"subHeading"} variation={"subdued"}>
        Sub-Heading
      </Heading>,
    ).toJSON();

    expect(view).toMatchSnapshot();
  });
});

describe("when Heading called with an alignment", () => {
  it("should match snapshot", () => {
    const view = render(
      <Heading align={"end"}>Text Aligned Right</Heading>,
    ).toJSON();

    expect(view).toMatchSnapshot();
  });
});

describe("when Heading called with maxLines", () => {
  it("should match snapshot", () => {
    const view = render(
      <Heading maxLines="single">Text Aligned Right</Heading>,
    ).toJSON();

    expect(view).toMatchSnapshot();
  });
});

describe("when Heading contains nested inline text", () => {
  it("renders nested Text within Heading", () => {
    const view = render(
      <Heading>
        Heading before <Text emphasis="strong">Inner</Text> after
      </Heading>,
    ).toJSON();

    expect(view).toMatchSnapshot();
  });
});
