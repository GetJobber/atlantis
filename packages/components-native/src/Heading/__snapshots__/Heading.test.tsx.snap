// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when Heading called with Subtitle variation should match snapshot 1`] = `
<Text
  accessibilityRole="header"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  maxFontSizeMultiplier={1}
  selectable={true}
  selectionColor="rgb(132, 234, 0)"
  style={
    [
      {
        "fontFamily": "inter-bold",
      },
      {
        "color": "rgb(1, 41, 57)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 24,
        "lineHeight": 28,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Subtitle
</Text>
`;

exports[`when Heading called with an alignment should match snapshot 1`] = `
<Text
  accessibilityRole="header"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  selectable={true}
  selectionColor="rgb(132, 234, 0)"
  style={
    [
      {
        "fontFamily": "inter-bold",
      },
      {
        "color": "rgb(1, 41, 57)",
      },
      {
        "textAlign": "right",
      },
      {
        "fontSize": 20,
        "lineHeight": 22,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Text Aligned Right
</Text>
`;

exports[`when Heading called with maxLines should match snapshot 1`] = `
<Text
  accessibilityRole="header"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  numberOfLines={1}
  selectable={true}
  selectionColor="rgb(132, 234, 0)"
  style={
    [
      {
        "fontFamily": "inter-bold",
      },
      {
        "color": "rgb(1, 41, 57)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 20,
        "lineHeight": 22,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Text Aligned Right
</Text>
`;

exports[`when Heading called with reverseTheme should match snapshot 1`] = `
<Text
  accessibilityRole="header"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  selectable={true}
  selectionColor="rgb(132, 234, 0)"
  style={
    [
      {
        "fontFamily": "inter-bold",
      },
      {
        "color": "rgba(255, 255, 255, 1)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 20,
        "lineHeight": 22,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Reverse Theme Heading
</Text>
`;

exports[`when Heading called with sub-heading variation and text-color should match snapshot 1`] = `
<Text
  accessibilityRole="header"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  maxFontSizeMultiplier={1.375}
  selectable={true}
  selectionColor="rgb(132, 234, 0)"
  style={
    [
      {
        "fontFamily": "inter-semibold",
      },
      {
        "color": "rgb(77, 105, 116)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Sub-Heading
</Text>
`;

exports[`when Heading called with sub-heading variation should match snapshot 1`] = `
<Text
  accessibilityRole="header"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  maxFontSizeMultiplier={1.375}
  selectable={true}
  selectionColor="rgb(132, 234, 0)"
  style={
    [
      {
        "fontFamily": "inter-semibold",
      },
      {
        "color": "rgb(1, 41, 57)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Sub-Heading
</Text>
`;

exports[`when Heading called with text as the only prop should match snapshot 1`] = `
<Text
  accessibilityRole="header"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  selectable={true}
  selectionColor="rgb(132, 234, 0)"
  style={
    [
      {
        "fontFamily": "inter-bold",
      },
      {
        "color": "rgb(1, 41, 57)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 20,
        "lineHeight": 22,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Default Heading
</Text>
`;

exports[`when Heading called with title variation should match snapshot 1`] = `
<Text
  accessibilityRole="header"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  maxFontSizeMultiplier={1.0625}
  selectable={true}
  selectionColor="rgb(132, 234, 0)"
  style={
    [
      {
        "fontFamily": "jobberpro-xbd",
      },
      {
        "color": "rgb(1, 41, 57)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 32,
        "lineHeight": 36,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Title Heading
</Text>
`;
